core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -249,14 +249,57 @@ class RequestSendThread(val controllerId: Int,
 
   private val socketTimeoutMs = config.controllerSocketTimeoutMs
 
+  private val controllerRequestMerger = new ControllerRequestMerger
+
+  private var firstUpdateMetadataWithPartitionsSent = false
+
+  def backoff(): Unit = pause(100, TimeUnit.MILLISECONDS)
+
   override def doWork(): Unit = {
+    val (requestBuilder, callback) = nextRequestAndCallback()
+    sendAndReceive(requestBuilder, callback)
+  }
 
-    def backoff(): Unit = pause(100, TimeUnit.MILLISECONDS)
+  private def nextRequestAndCallback(): (AbstractControlRequest.Builder[_ <: AbstractControlRequest], AbstractResponse => Unit) = {
+    if ((!config.liCombinedControlRequestEnable) || (!firstUpdateMetadataWithPartitionsSent)) {
```

Comment:
check if there is no merging requests before going back to single request send model?

Reply:
The check has been added.